docs/current/guides/281969-dagger-on-kubernetes.md

```diff
@@ -0,0 +1,218 @@
+
+1. **Create a set of taints and tolerations for those runners**: Taints and tolerations in Kubernetes are a way to ensure that certain nodes are reserved for specific tasks. By setting up taints on your runner nodes, you can prevent other workloads from being scheduled on them. Tolerations are then added to the runners so that they can be scheduled on these tainted nodes. This ensures that your runners have dedicated resources for their tasks.
+
+1. **Set up the same taints and tolerations for a Dagger Engine DaemonSet**: A DaemonSet ensures that all (or some) nodes run a copy of a Pod. In this case, you want to ensure that a Dagger Engine is running on each of your runner nodes. By setting up the same taints and tolerations as your runners, you ensure that the Dagger Engines are co-located with your runners.
+
+1. **Use Karpenter for auto-scaling of node groups**: Karpenter is a Kubernetes-native node autoscaler. It can automatically adjust the size of your node groups based on the current workload. This means that if you have a lot of CI jobs running, Karpenter can automatically add more nodes to your cluster. Conversely, if there are few jobs running, it can remove unnecessary nodes, saving you money.
+
+1. **Choose AWS nodes with moderate to large NVMe drives for the Dagger Engine cache**: The Dagger Engine cache is used to store intermediate build artifacts, which can significantly speed up your CI jobs. By choosing nodes with large NVMe drives, you ensure that there is plenty of space for this cache. NVMe drives are also much faster than traditional SSDs, which can further improve performance.
+
+Remember, these steps may vary depending on your specific setup and requirements. Always refer to the official documentation of the tools you are using for the most accurate and up-to-date information.
+
+### Dagger Engine DaemonSet
+
+```yaml
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: dagger-engine
+  namespace: dagger
+  labels:
+    k8s-app: dagger-engine
+spec:
+  selector:
+    matchLabels:
+      name: dagger-engine
+  template:
+    metadata:
+      labels:
+        name: dagger-engine
+    spec:
+      tolerations:
+        - effect: NoSchedule
+          key: github-actions-runner
+          operator: Exists
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: github-actions-runner
+                operator: Exists
+      securityContext:
+        runAsUser: 0
+        runAsGroup: 1001
+        fsGroup: 1001
+        fsGroupChangePolicy: "OnRootMismatch"
+      initContainers:
+        - name: download-bin
+          image: {{ .Values.dagger.engine.image.repository }}:{{ .Values.dagger.engine.image.tag }}
```

Comment:
You might either want to link to a Helm chart or provide a static file. IME it seems to be an even spread between folks using Kustomize and Helm.

Reply:
There's a PR in progress for a Helm chart here too: #5505